docs/source/reference/types.rst

```diff
@@ -100,6 +100,118 @@ the beginning or the end of the index specification::
    >>> numba.float32[::1, :, :]
    array(float32, 3d, F)
 
+Functions
+---------
+
+.. warning::
+   The feature of considering functions as first-class type objects is
+   under development.
+
+While functions are often considered as certain transformations of
+input arguments to output values then within Numba jit compiled
+functions, the functions can also be considered as objects, that is,
+functions can be passed around as arguments or return values, or used
+as items in sequences, in addition of being callable.
+
+To enable first-class function support, Numba :term:`JIT` compiled
+functions must specify ``firstclass=True`` ``jit`` option. Numba
```

Comment:
The docs need to be updated. `firstclass` option is removed in this PR.

Reply:
Fixed.